CVLByExample/summary/with-env/WithEnvCVLFunctionSummary/withEnvSummary.spec

```diff
@@ -0,0 +1,52 @@
+/**
+  An example for using a summary with `with env`.
+*/
+
+using Basic as basic;
+
+methods {
+  // Summarization by a CVL function. The result is non-deterministic because the CVL function is not deterministic.
+  function reduceBalance(uint256 amount) internal returns (uint256) with (env e) => cvlReduceBalance(e, amount);
+  function totalSupply() external returns (uint256) envfree;
+}
+
+// CVL function used for summarization. Simplifies the summarized function.
+function cvlReduceBalance(env e, uint256 amount) returns uint256  {
```

Comment:
this doesn't look simplified, if you just want non determinstic, just declare a cvl variable and return it